google/resumable_media/_helpers.py

```diff
@@ -192,17 +192,17 @@ def wait_and_retry(func, get_status_code, retry_strategy):
         else:
             return response
 
-        if not retry_strategy.retry_allowed(total_sleep, num_retries):
-            # Retries are exhausted and no acceptable response was received.
-            # Raise the retriable_error.
-            raise error
-
         base_wait, wait_time = calculate_retry_wait(
             base_wait, retry_strategy.max_sleep, retry_strategy.multiplier
         )
-
         num_retries += 1
         total_sleep += wait_time
+
+        # Check if (another) retry is allowed. If retries are exhausted and
+        # no acceptable response was received, raise the retriable error.
+        if not retry_strategy.retry_allowed(total_sleep, num_retries):
```

Comment:
The docstring for `retry_strategy.retry_allowed()` says that the num_retries is supposed to be the number of retries used so far. After thinking about it, I agree with your changes here and think that docstring is wrong. We should modify the docstring to reflect this change.

Reply:
@andrewsg I've updated the docstring and added another test case. PTAL, thanks!